transaction.go

```diff
@@ -407,7 +409,10 @@ func (t *Transaction) Encode() []byte {
 // able to decode outputs from PayloadMessage(), EnvelopeMessage() and Encode()
 // functions
 func DecodeTransaction(transactionMessage []byte) (*Transaction, error) {
-	temp, err := decodeTransaction(transactionMessage)
+
+	transasctionMessageWithoutDomainTag := bytes.TrimPrefix(transactionMessage, TransactionDomainTag[:])
```

Comment:
I trying to think of the best place to add the domain tag. I'm wondering if it should go directly in the transaction message (like you have it) or be handled by the signer instead. 

- The benefit of doing it in the signer is that it removes the need to handle cases like this.
- The benefit of doing it in the transaction is that it ensure that nobody forgets to add the domain tag (e.g. if using a completely custom signing implementation).

I'll have to think about it more -- my brain is a bit tired haha